src/getAddress.c

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <os.h>
+#include <os_io_seproxyhal.h>
+#include <string.h>
+#include "hathor.h"
+#include "ux.h"
+
+static getAddressContext_t *ctx = &global.getAddressContext;
+
+// Define the comparison screen. This is where the user will compare the address on
+// their device to the one shown on the computer. There are lef/right buttons on the
+// screen to navigate the hash. Clicking both buttons goes back to main screen.
+static const bagl_element_t ui_getAddress_compare[] = {
+    UI_BACKGROUND(),
+
+    // Left and right buttons for scrolling the text.
+    UI_ICON_LEFT(0x01, BAGL_GLYPH_ICON_LEFT),
+    UI_ICON_RIGHT(0x02, BAGL_GLYPH_ICON_RIGHT),
+
+    UI_TEXT(0x00, 0, 12, 128, "Address:"),
+    UI_TEXT(0x00, 0, 26, 128, global.getAddressContext.partialAddress),
+};
+
+// Preprocessor for this screen. Hides left or right arrows depending on the
+// scrolling position.
+static const bagl_element_t* ui_prepro_getAddress_compare(const bagl_element_t *element) {
+    switch (element->component.userid) {
+    case 1:
+        // 0x01 is the left icon so return NULL if we're displaying the beginning of the text.
+        return (ctx->displayIndex == 0) ? NULL : element;
+    case 2:
+        // 0x02 is the right, so return NULL if we're displaying the end of the text.
+        return (ctx->displayIndex == sizeof(ctx->b58_address)-12) ? NULL : element;
```

Comment:
Why do you subtract 12? Is it the screen size? Maybe we should move it to a constant.

Reply:
Yes, it's the maximum string size we show on the screen.

I'll move it to a define on one of the header files.

Reply:
Defined this on ux.h